### PR TITLE
Use `touch` instead of `update_all` on option_type -> product touch.

### PR DESCRIPTION
**Pull request by philbirt**
To allow the product to touch other related objects. Otherwise the touching chain is halted here.

Also make sure that touch gets called on save.

### CODE REVIEW

**Comment by athal7** [comment]
:+1:

File: core/app/models/spree/option_type.rb [hunk 1]
```diff
@@ -21,10 +21,11 @@ class OptionType < Spree::Base
 
     accepts_nested_attributes_for :option_values, reject_if: lambda { |ov| ov[:name].blank? || ov[:presentation].blank? }, allow_destroy: true
 
+    after_save :touch
```
Comment (magnusvk) [inline core/app/models/spree/option_type.rb]: This seems weird. Doesn't that just trigger another, unnecessary update.

Can we either:
a) remove it?
b) if that doesn't work just do `after_save touch_all_products`?

**Comment by jhawthorn** [comment]
:+1: but we do need to revisit touching.

**Comment by magnusvk** [comment]
Also would be nice to have a spec that tests whether this works to make it easy to verify whether we need the `after_save :touch`.

**Comment by athal7** [comment]
i actually don't think it is necessary. removed

**Comment by philbirt** [comment]
Actually the after_save hook is necessary, but updated it per @magnusvk comments

**Comment by magnusvk** [comment]
So, not to be difficult or anything, but if it _is_ necessary and it's not easily apparent to us as to why, then it should have a spec that fails if it's not there.

**Comment by gmacdougall** [comment]
Is there any reason why we can't just declare `touch: true` on the `has_many :products`? I think the touch_all_products was implemented to (badly) try and update the products quickly.

**Comment by athal7** [comment]
@gmacdougall I don't believe that's an option on has_many, just on belongs_to http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html

**Comment by gmacdougall** [comment]
Of course. Thanks for that.

:+1:

**Pull request merged**